packages/server-common/src/handlers/user/sync-active-campaign-contact-factory.ts

```diff
@@ -0,0 +1,136 @@
+import { NotFoundError } from '@asap-hub/errors';
+import {
+  UserEvent,
+  gp2,
+  UserResponse,
+  UserUpdateRequest,
+} from '@asap-hub/model';
+import { isBoom } from '@hapi/boom';
+import { EventBridgeEvent } from 'aws-lambda';
+
+import { UserPayload } from '../event-bus';
+import {
+  EventBridgeHandler,
+  Logger,
+  ContactPayload,
+  ContactResponse,
+} from '../../utils';
+
+export interface UserController {
+  fetchById(id: string): Promise<gp2.UserResponse | UserResponse>;
+  update(
+    id: string,
+    update: UserUpdateRequest | gp2.UserUpdateRequest,
+  ): Promise<gp2.UserResponse | UserResponse>;
+}
+
+export const syncActiveCampaignContactFactory =
+  <Controller extends UserController>(
```

Comment:
are you sure you need this generic here? can you not just expect userController which implements the UserController interface?

Reply:
Yeah, I think this is not necessary, I'll remove it :)